typing/typedecl.ml

```diff
@@ -1567,7 +1598,7 @@ let transl_type_extension extend env loc styext =
   | None -> ()
   | Some err -> raise (Error(loc, Extension_mismatch (type_path, env, err)))
   end;
-  let ttype_params = make_params env styext.ptyext_params in
+  let ttype_params = make_params env type_path styext.ptyext_params in
```

Comment:
This `type_path` is potentially confusing:

```ocaml
type 'a t = ..
type ('a : immediate) t += Foo
```

The output will make it look like the problem is on the first line, not the second.... but these lines might even be in different files! This would be solved if we can store the source location of the definition.